web/src/app/selection/MaterialSelect.js

```diff
@@ -1,10 +1,148 @@
-import React, { Component } from 'react'
+import React, { useState } from 'react'
 import { PropTypes as p } from 'prop-types'
-import { withStyles } from '@material-ui/core/styles'
-import Select from 'react-select'
-import { components, styles } from './MaterialSelectComponents'
-import shrinkWorkaround from '../util/shrinkWorkaround'
-import _ from 'lodash-es'
+import {
+  TextField,
+  makeStyles,
+  MenuItem,
+  ListItemIcon,
+  Typography,
+  Paper,
+  Chip,
+} from '@material-ui/core'
+import { Autocomplete } from '@material-ui/lab'
+
+const useStyles = makeStyles({
+  listItemIcon: {
+    minWidth: 0,
+  },
+  menuItem: {
+    display: 'flex',
+    flex: 1,
+    justifyContent: 'space-between',
+    wordBreak: 'break-word',
+    whiteSpace: 'pre-wrap',
+  },
+  option: {
+    padding: 0,
+  },
+  clearIndicator: {
+    display: 'none',
+  },
+})
+
+export default function MaterialSelect(props) {
+  const classes = useStyles()
+  const {
+    canCreate,
+    disabled,
+    error,
+    isLoading,
+    label,
+    multiple,
+    name,
+    noOptionsText,
+    onChange,
+    onInputChange,
+    options,
+    placeholder,
+    required,
+    value: propsValue,
+  } = props
+
+  let value = propsValue
+  if (!value) {
+    if (multiple) value = []
+    else value = { label: '', value: '' }
+  }
+
+  const [inputValue, setInputValue] = useState(multiple ? '' : value.label)
+
+  return (
+    <Autocomplete
+      data-cy='material-select'
+      data-cy-ready={!isLoading}
+      classes={{
+        option: classes.option,
+        clearIndicator: classes.clearIndicator,
+      }}
+      value={multiple ? value : value.value}
+      inputValue={inputValue}
+      disableClearable={required}
+      disabled={disabled}
+      multiple={multiple}
+      filterSelectedOptions
+      noOptionsText={noOptionsText}
+      onChange={(event, valueObj) => {
+        onChange(valueObj)
+
+        if (valueObj !== null) {
+          let newInputVal = ''
+          if (!multiple) {
+            if (canCreate && options.length === 1) {
+              newInputVal = options[0].value
+            } else {
+              newInputVal = valueObj.label
+            }
+          }
+          setInputValue(newInputVal)
+        }
```

Comment:
- fire `onChange` handlers after input has been handled rather than before
- let's flatten this somehow
- what happens with 1 valid search result (not created) and `canCreate` is true? Can we set a flag on created options instead?